etc/scripts/get-stack.sh

```diff
@@ -726,9 +726,14 @@ has_cmd() {
   command -v "$1" > /dev/null 2>&1
 }
 
-# Check whether the given path is listed in the PATH environment variable
+# Check whether the given (query) path is listed in the PATH environment variable.
 on_path() {
-  echo ":$PATH:" | grep -q :"$1":
+    # Below we normalize PATH and query by ensuring ~ is expanded to $HOME, avoiding
```

Comment:
POSIX doesn't specify tilde-expansion for variables in PATH. That's implementation specific behavior. E.g. in dash shell `~/foo/bin` in PATH is treated as a relative directory.

The only real way to detect whether a directory is listed in PATH is to drop a dummy script in that directory and try to invoke it by filename. But that's gonna be problematic if we don't have write permissions. But I believe the stack script would fail for other reasons if those write permissions aren't in place.

Reply:
I didn't know that expansion of ~ in PATH was not POSIX, thanks for sharing that!
In that case I will close the PR.